src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java

```diff
@@ -100,12 +103,11 @@ private static SymbolLookup makeAixLookup() {
                  */
                 SymbolLookup funcsLibLookup =
                         libLookup(libs -> libs.load(jdkLibraryPath("syslookup")));
-                MemorySegment funcs = MemorySegment.ofAddress(funcsLibLookup.find("funcs").orElseThrow().address(),
-                    ADDRESS.byteSize() * AixFuncSymbols.values().length, SegmentScope.global());
+                MemorySegment funcs = MemorySegment.ofAddress(funcsLibLookup.find("funcs").orElseThrow().address());
```

Comment:
Suggest this should follow the pattern used for Windows:
```java
                MemorySegment funcs = funcsLibLookup.find("funcs").orElseThrow()
                        .reinterpret(AixFuncSymbols.LAYOUT.byteSize());
```
This, of course, requires a definition of `AixFuncSymbols.LAYOUT`:
```java
        static final SequenceLayout LAYOUT = MemoryLayout.sequenceLayout(
                values().length, ADDRESS);
```

Reply:
Agreed and updated the code as suggested above.